src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c

```diff
@@ -20,121 +104,2126 @@ const ActorInit Demo_Effect_InitVars = {
+    f32 xzSpherical;
+
+    distance = frameDivisor * sqrtf(SQ(endPos.x - startPos.x) + SQ(endPos.y - startPos.y) + SQ(endPos.z - startPos.z));
+
+    this->actor.posRot.pos.x = radius * cosf(degrees * (M_PI / 180.0f));
+    this->actor.posRot.pos.y = distance;
+    this->actor.posRot.pos.z = radius * sinf(degrees * (M_PI / 180.0f));
+
+    xPos = this->actor.posRot.pos.x;
+    ySpherical = (this->actor.posRot.pos.y * cosf(rotation.x * (M_PI / 32768.0f))) -
+                 (sinf(rotation.x * (M_PI / 32768.0f)) * this->actor.posRot.pos.z);
+    xzSpherical = (this->actor.posRot.pos.z * cosf(rotation.x * (M_PI / 32768.0f))) +
+                  (sinf(rotation.x * (M_PI / 32768.0f)) * this->actor.posRot.pos.y);
+
+    this->actor.posRot.pos.x =
+        (xPos * cosf(rotation.y * (M_PI / 32768.0f))) - (sinf(rotation.y * (M_PI / 32768.0f)) * xzSpherical);
+    this->actor.posRot.pos.y = ySpherical;
+    this->actor.posRot.pos.z =
+        (xzSpherical * cosf(rotation.y * (M_PI / 32768.0f))) + (sinf(rotation.y * (M_PI / 32768.0f)) * xPos);
+
+    this->actor.posRot.pos.x += startPos.x;
+    this->actor.posRot.pos.y += startPos.y;
+    this->actor.posRot.pos.z += startPos.z;
+}
+
+/**
+ * Moves the Jewel Actor spherically from startPos to endPos.
+ * This is used by the Jewel Actor during the Door of Time activation cutscene.
+ * This is run when the Jewels merge from Link and begin orbiting him.
+ */
+void DemoEffect_MoveJewelActivateDoorOfTime(DemoEffect* this, GlobalContext* globalCtx) {
+    Vec3f startPos;
+    Vec3f endPos;
+    f32 frameDivisor;
+    f32 degrees;
+    f32 radius;
+    s32 csActionId;
+
+    csActionId = this->csActionId;
+    startPos.x = globalCtx->csCtx.npcActions[csActionId]->startPos.x;
+    startPos.y = globalCtx->csCtx.npcActions[csActionId]->startPos.y;
+    startPos.z = globalCtx->csCtx.npcActions[csActionId]->startPos.z;
+    endPos.x = globalCtx->csCtx.npcActions[csActionId]->endPos.x;
+    endPos.y = globalCtx->csCtx.npcActions[csActionId]->endPos.y;
+    endPos.z = globalCtx->csCtx.npcActions[csActionId]->endPos.z;
+
+    frameDivisor = DemoEffect_InterpolateCsFrames(globalCtx, csActionId);
+
+    switch (this->jewel.type) {
+        case Demo_Effect_Jewel_Kokiri:
+            degrees = 0.0f;
+            break;
+        case Demo_Effect_Jewel_Goron:
+            degrees = 120.0f;
+            break;
+        case Demo_Effect_Jewel_Zora:
+            degrees = 240.0f;
+            break;
+    }
+
+    radius = 50.0f * frameDivisor;
+    // CLAMP_MAX macro is not OK
+    if (radius > 30.0f) {
+        radius = 30.0f;
+    }
+
+    if (startPos.x != endPos.x || startPos.y != endPos.y || startPos.z != endPos.z) {
+        this->jewelCsRotation.x = atan2f(endPos.z - startPos.z, -(endPos.x - startPos.x)) * (32768.0f / M_PI);
+        this->jewelCsRotation.y = Math_Vec3f_Yaw(&startPos, &endPos);
+    }
+
+    this->jewelCsRotation.z += 1024;
+
+    degrees += this->jewelCsRotation.z * (45.0f / 8192.0f);
```

Comment:
this makes a little more sense as `360.0f / 0x10000`

Reply:
Why not `360 / 65536`?
Agreed that including 360 in the number if better.